src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -1356,14 +1358,16 @@ static int CheckEventJSON(cJSON *event, cJSON **scan_id, cJSON **id, cJSON **nam
 
     } else {
 
-        if( *id = cJSON_GetObjectItem(*check, "id"), !*id) {
+        if ((*id = cJSON_GetObjectItem(*check, "id")) == NULL)
+        {
             merror("Malformed JSON: field 'id' not found.");
             return retval;
         }
 
         obj = *id;
-        if( !obj->valueint ) {
-            merror("Malformed JSON: field 'id' must be a string.");
+        if (!cJSON_IsNumber(obj))
```

Comment:
Please, let's keep the typical format we use in the rest of the functions, with the bracket on the same line as the header.